### DISCUSSION TITLE
How to reset properties using names, not indexes?

### DISCUSSION THREAD

**Discussion by bryanwhiting** [h2oai/wave#1779] [Q&A] [answered]
H2O wave noob here (only 5 hours in so far, but loving it!)

Per https://wave.h2o.ai/docs/examples/form-visibility, the code to set the visibility status is complicated, requiring you to know the index location of the element:

```
items = q.page['example'].items
    items_to_hide = [
        items[0].text_xl,
        items[2].text_m,
        items[4].inline.items[0].button,
        items[5].buttons.items[2].button,
    ]
    if q.args.hide:
        for i in items_to_hide:
            i.visible = False
    if q.args.show:
        for i in items_to_hide:
            i.visible = True
```

for example, if I set the "name" property from 
```
ui.text_xl(content='First text'),
```
to 
```
ui.text_xl(content='First text', name='first_text),
```

I'd love to be able to just do:
```
 items.first_text.text_xl.visible=False
#or
 items['first_text'].text_xl.visible=False
```

I know items is a list. but why isn't it designed as a dictionary? I feel like this would make the names much more useful and the code much more readable.

**Comment by bryanwhiting** [comment]
Something like this would be great:
```
...
ui.text_xl(name='text1', content='First text'),
 ui.text_l(name='text2', content='Second text'),
...
items_to_hide = [i for i in items if i.name in ('text1','text2')]
```
but admittedly this might reflect my deep lack of understanding of how wave stores metadata. would love to better understand!

**Reply by mturoci** [reply]
The python UI objects in Wave are write-only. Notice that you get an Expando instance when trying to print the existing cards.

See my other comment for how we plan to address this.

**Comment by mturoci** [comment, marked as answer]
Hello @bryanwhiting, you are right, the current API is cumbersome. Will be resolved by https://github.com/h2oai/wave/issues/148 hopefully in the near future :) Thanks for the feedback, much appreciated!

**Reply by bryanwhiting** [reply]
thanks for building an awesome app! in just 5 hours (from bare bones not having coded in h2o-wave at all) i was able to build a simple app that could visualize 3 target variables across 600 datasets with helpful dropdowns. can't wait to see what happens now that I know what's going on!

**Reply by mturoci** [reply]
That's awesome! Keep up the great work :)
